### ISSUE TITLE
test result

### ISSUE THREAD

**Issue by miao02830** [voldemortX/pytorch-auto-drive#28]
the step 2:
predict and save lanes 
python main_landec.py --state=<state> \  # 2: test set; 3: validation set           
                      --continue-from=<path to .pt file> \
                      --dataset=<dataset> \ 
                      --method=<the method used> \
                      --backbone=<the backbone used> \ 
                      --batch-size=<any batch size> \  # Recommend 80
                      --mixed-precision  # Enable mixed precision
i run it without any return. where the result save? ths

**Comment by voldemortX** [comment]
If it did not raise any errors (you have seen the tqdm progress bar finish), then you may find predicted lane lines (.txt files) in `./output`.

**Comment by miao02830** [comment]
python main_landec.py --state= \ # 2: test set; 3: validation set
--continue-from=<path to .pt file>
--dataset= \
--method=
--backbone= \
--batch-size= \ # Recommend 80
--mixed-precision # Enable mixed precision

i run it and it shows that i finished test 
![IMG_3712](https://user-images.githubusercontent.com/27732256/133869966-a534097d-220d-4bba-965a-5330a3d6a0b2.JPG)
but i didn't get any result . i dont'have the file ./output
![IMG_3713](https://user-images.githubusercontent.com/27732256/133870036-fde682ef-b7d0-4190-bfb4-c96d5b171912.JPG)

just get the output under root,but it didn't have any info.
![0731DDDC8C69C968AADC21970966F207](https://user-images.githubusercontent.com/27732256/133870074-156e5b8c-ed5d-495e-9320-f23fae0da970.png)
i.m very confused.thank you very much.

**Comment by voldemortX** [comment]
That is weird indeed, can I have your full shell command?

**Comment by voldemortX** [comment]
In my machine, CULane test results are txt files and TuSimple result is a single json file.

**Comment by miao02830** [comment]
> In my machine, CULane test results are txt files and TuSimple result is a single json file.

I think so, too but i don't have any results.

this is my full test shell:
python main_landec.py --state='2' --dataset='culane' --continue-from='erfnet_baseline_culane.pt' --method='baseline'  --backbone='erfnet'  --batch-size='8'  --mixed-precision
 
and it didn't raise any errors and shows that test finished. but don't have any output . thankyou for your replay.

**Comment by voldemortX** [comment]
@miao02830 I'll try reproduce this on my machine. In the meantime, can you double-check your `output` folder in the command line with `ls` , and check if your dataset is setup correct? Also how long did the inference run, did it run with ~4000 iters?

**Comment by miao02830** [comment]
ok thankyou! I'll check them one by one，and  look forward to your reply!

**Comment by miao02830** [comment]
> @miao02830 I'll try reproduce this on my machine. In the meantime, can you double-check your `output` folder in the command line with `ls` , and check if your dataset is setup correct? Also how long did the inference run, did it run with ~4000 iters?
> 
> p.s. It is possible that a bug is introduced in recent refactors. [288e395](https://github.com/voldemortX/pytorch-auto-drive/commit/288e3959f2529266065458718de94f4f78b8c2d1) maybe also try the latest stable release `git checkout v1.4`

I checked the path of my dataset.  i use 'ls' to check that the output is empty. I hope you can tell me in which file and in which statement the output result is written? Moreover, I don't quite understand how to check "also how long did the information run, did it run with ~ 4000 iters?" I hope you can give me some advice. Thank you

**Comment by voldemortX** [comment]
at the end of that progress bar, there should be a runtime and iter count?

**Comment by miao02830** [comment]
Oh, I see. After the previous run, I forgot the screenshot. I remember that it ran for about 40 minutes. Like this, a total of 4335, up to 100%.
![IMG_3745](https://user-images.githubusercontent.com/27732256/133879948-5cf0ecfb-d215-4087-a6de-fe1fdbea3e35.JPG)

**Comment by voldemortX** [comment]
@miao02830 I tested your shell command on my machine, I can get output txt normally, they should look like this:

./output:
- driver_100_30frame
    -  05251551_0440.MP4
        - 04170.lines.txt
    - ...
- ...

Did you run the newest code from master and prepared your culane dataset with [these instructions](https://github.com/voldemortX/pytorch-auto-drive/blob/master/docs/datasets/CULANE.md) ?

**Comment by miao02830** [comment]
> @miao02830 I tested your shell command on my machine, I can get output txt normally, they should look like this:
> 
> ./output:
> 
> * driver_100_30frame
>   
>   * 05251551_0440.MP4
>     
>     * 04170.lines.txt
>   * ...
> * ...
> 
> Did you run the newest code from master and prepared your culane dataset with [these instructions](https://github.com/voldemortX/pytorch-auto-drive/blob/master/docs/datasets/CULANE.md) ?

thankyou very much. I got it! i just rename 'list' as 'lists',not do it 
![image](https://user-images.githubusercontent.com/27732256/133915678-90dfc169-fd4e-425a-9ec0-a67423c407a0.png)

thankyou!!

but i also have one question。
i got the result and the dataset is right.but when i evaluate,i just get the tp,the fp and fn is 0.

![IMG_3765](https://user-images.githubusercontent.com/27732256/133915760-a514efb6-04e0-4fdf-8193-61c8ea1c5410.JPG)
 
i checked my data_dir and detect_dir ,and it shows right. 

tks!

**Comment by voldemortX** [comment]
maybe you should also run the culane_list_convertor.py? it seems you just tested gt anootation against gt annotation...

**Comment by miao02830** [comment]
It seems so from the result. 
![IMG_3766](https://user-images.githubusercontent.com/27732256/133916245-59694555-b5c8-421b-9348-79b38c34594d.JPG)

But when I check the code, it should be that   two parts are compared.
![image](https://user-images.githubusercontent.com/27732256/133916335-391be28d-6499-48d2-912f-60565c8568d8.png)
Is my understanding right?

tks

**Comment by voldemortX** [comment]
> It seems so from the result. 
> ![IMG_3766](https://user-images.githubusercontent.com/27732256/133916245-59694555-b5c8-421b-9348-79b38c34594d.JPG)
> 
> But when I check the code, it should be that   two parts are compared.
> ![image](https://user-images.githubusercontent.com/27732256/133916335-391be28d-6499-48d2-912f-60565c8568d8.png)

It does look correct (if it is gt you should not see the 1 fp. ) Did you test from our uploaded weights or trained your own network?

**Comment by miao02830** [comment]
i test from your uploaded weights.i retest it again but it also shows fp is 1. it makes me confused...

**Comment by voldemortX** [comment]
> i test from your uploaded weights.i retest it again but it also shows fp is 1. it makes me confused...

Then it should show the reported accuracies.  Maybe try clone the repo again, delete `lists` and prepare your dataset again step by step?

**Comment by miao02830** [comment]
Thank you very much. OK, I'll try again from the beginning.

**Comment by voldemortX** [comment]
@miao02830 It seems the problem is somewhat resolved and happened a long time ago. Please open a new issue if the problem continues to show up in the new master branch.
